ambari-server/src/main/resources/common-services/HBASE/2.0.0.3.0/package/scripts/params_linux.py

```diff
@@ -184,6 +184,9 @@
 user_group = config['configurations']['cluster-env']["user_group"]
 
 if security_enabled:
+  zk_principal_name = default("/configurations/zookeeper-env/zookeeper_principal_name", "zookeeper/_HOST@EXAMPLE.COM")
```

Comment:
I just have a small background in these tasks,  but are you guys sure the zookeeper principal name should be used here? Its not clean based on the property name. If a znode is secured, and only let hbase (or hbase-<clustername>) sasl user to access the znode, maybe on client side we would need to use that one instead of zookeeper principal. (as in the name that is a client, it sounds more like that)

Reply:
@oleewere , When accessing a service, the client needs to request a service ticket by asking the KDC for one by supplying its own identity and the principal of the service it wants to access.  In this case, the Zookeeper library, by default assumes that the Zookeeper principal is `zookeeper/_HOST@REALM`.  If the service principal for the Zookeeper service has be set to something different, then the client will create a request for the wrong or an invalid service.  By explicitly obtaining the Zookeeper service principal from the configurations, that service principal can specified rather than allow the client assume the default value. 

Since the Zookeeper service principal is rarely changed, this has not been an issue.  But in case someone wants to change it, the ability should be available.